[docs-infra] Adjust the `kbd` tag styles

danilo-leal opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

This PR adds some styling tweaks to the `<kbd>` tag and adds instructions about using the specific `key` class for them. Initially, this PR changed the Markdown Element file targeting to get the whole tag, but I learned through Alex (down below) that the class existed for when we want to wrap individual tags with a parent `kbd` to indicate that they should be pressed together. 

👉 **https://deploy-preview-39397--material-ui.netlify.app/experiments/docs/markdown/#kbd-tag**

alexfauquette commented:
I think we do need it 😁

https://deploy-preview-39397--material-ui.netlify.app/blog/docs-restructure-2022/

![image](https://github.com/mui/material-ui/assets/45398769/e3077845-303f-47aa-8eda-505bc05499aa)

I think it comes from this article https://developer.mozilla.org/en-US/docs/Web/HTML/Element/kbd#representing_keystrokes_within_an_input

Where they use `<kbd><kbd>Ctrl</kbd>+<kbd>N</kbd></kbd>` to mean that ctrl and N are two keys, and they are pressed together.

A solution could be to use `kbd:not(:has(kbd))` by `:has()` is supported by only [87% of the browser](https://caniuse.com/?search=%3Ahas())

Maybe the easier is to fix the few occurrences of `<kbd` to add class="key" when missing and send a slack message explaining why its needing such that every body get the hint

alexfauquette commented on docs/pages/experiments/docs/markdown.md:
```diff
@@ -34,3 +34,9 @@ https://spec.commonmark.org/0.30/#loose
+
+**In isolation:** <kbd class="key">caps lock</kbd>
+
+**In context:** The following demo shows how the `focusableWhenDisabled` prop works—use the <kbd>Tab</kbd> key to navigate within this document to see that only the second Button accepts the focus:
```
```suggestion
**In context:** The following demo shows how the `focusableWhenDisabled` prop works—use the <kbd>Tab</kbd> key to navigate within this document to see that only the second Button accepts the focus.
It's also important to support shotcuts such as <kbd><kbd class="key">Ctrl</kbd>+<kbd class="key">C</kbd></kbd>. Notice the use of `class='key'` to distinguish the shortcut and its keys
```

danilo-leal commented:
@alexfauquette Ohh... gotcha, that makes sense! Thanks for calling that out! I rolled back the class declaration, added instructions on the Markdown page, and fixed the missing instances! 🤙

alexfauquette reviewed (approved):
Nice :)
